### PR TITLE
[crypto] Extra functional tests for threshold sig verification

### PR DESCRIPTION
**Pull request by kchalkias**
## Motivation

Creating extra functional tests for threshold signature verification corner cases.
Actually, the tests revealed a bug according to which the key size was not properly checked.
The PR solves this problem too. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

YES

## Test Plan

```
threshold_signature_ed25519_bad_msg.mvir
threshold_signature_ed25519_bad_pk.mvir
threshold_signature_ed25519_bad_sig.mvir
threshold_signature_ed25519_bad_sig_size.mvir

threshold_signature_ed25519_no_matching_pk.mvir
threshold_signature_ed25519_oversized_bitmap.mvir
threshold_signature_ed25519_undersized_bitmap.mvir
threshold_signature_ed25519_oversized_pk.mvir
threshold_signature_ed25519_zero_bitmap_failure.mvir

```
## Related PRs

extends work on [#335](https://github.com/libra/libra/pull/335)

### CODE REVIEW

File: language/stdlib/natives/src/signature.rs [hunk 1]
```diff
@@ -156,23 +156,46 @@ fn sanity_check(bitmap: &BitVec<u32>, signatures: &ByteArray, pubkeys: &ByteArra
...
     }
     // Ensure we have as many signatures as the number of set bits in bitmap.
     if bitmap_count_ones * 64 != signatures_len {
-        bail!("Mismatch between Bitmap Hamming weight and number of signatures");
+        return Err(ThresholdSignatureError::SignatureSizeFailure.into());
     }
     // Ensure that we have at least as many keys as the index of the last set bit in bitmap.
-    if public_keys_len < 32 * bitmap_last_bit_set {
-        bail!("Bitmap points to a non-existent key");
+    if public_keys_len < 32 * (bitmap_last_bit_set + 1) {
```
Comment (kchalkias) [inline language/stdlib/natives/src/signature.rs]: FYI, this was the failing check.

**Comment by huitseeker** [comment]
Darn. And I'm juuust about to change the format of signatures >_<

**Comment by kchalkias** [comment]
@huitseeker, no worries, feel free to do and I’ll rebase

**Comment by kchalkias** [comment]
@huitseeker shall I wait for your conflicting PR, then rebase before I merge?
